az-digital/az_quickstart
# Closes #556 Creating custom flexible block (new and improved: fixing branching problems)

This PR creates a custom block type to match QS1 flexible block.

## Description
- This PR add a custom block type called Flexible Block that includes the same paragraphs as Page
- Also creates a new module az_flexible_block where the associated files are stored)
![image](https://user-images.githubusercontent.com/10873398/112060817-e312c200-8b1a-11eb-9c4b-5daa26ef9711.png)

Further improvements needed (either on this PR or future tickets):
-Other fields to add? QS 1 has label, title, revision settings, view mode, and URL redirects, along with Bean stuff I don't totally understand)
-Make block viewable when you click on it from the list (currently goes to edit screen)

## Related Issue
#556

## How Has This Been Tested?
Tested locally and on Probo build

How to test: 

First create a flexible block:
1. Create a flexible block by going to Structure > Block Layout > Custom Block Library and then click the +Add Custom Block button
2. Choose Flexible Block
3. Enter a block description (should not display)
4. Add paragraphs with content
5. Save the block

Embed the block on a page:
1. Create a new page
2. Add a text area
3. Add a custom block using the E button on the editor toolbar
4. Choose the flexible you created
5. Save the page and make sure the block appears as expected

Or place the block:
1. Go to Structure > Block Layout
2. Choose a region and click the Place Block button
3. Select the flexible block you created
3. Specify the page/content type the block should/should not appear on (default would be all pages)
4. Make sure the block appears as expected

## Types of changes


## Checklist:
